### PR TITLE
Snowflake Source

### LINKED ISSUE #652: Feature Request: Snowflake Source
**Is your feature request related to a problem? Please describe.**
Add Snowflake as a source to import data into Grouparoo properties.

### PR DESCRIPTION
**Pull request by bleonard**
Connect to snowflake to pull in data.

Issue https://github.com/grouparoo/grouparoo/issues/652

### CODE REVIEW

**Comment by bleonard** [comment]
Also refactored SQL connection code to share amongst big query, postgres, redshift, mysql, and friends.

**Comment by bleonard** [comment]
Close T-539

File: plugins/@grouparoo/redshift/README.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
-# @Grouparoo/Redshift
+# @Grouparoo/Snowflake
```
Comment (evantahler) [inline plugins/@grouparoo/redshift/README.md]: Looks like you changed the wrong file

File: plugins/@grouparoo/snowflake/__tests__/utils/nockHelper.ts [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      // somehow, these bytes are slightly different (1 hex digit) on mac/ci
+      // mac: 1f8b0800000000000013558...
+      //  ci: 1f8b0800000000000003558...
+      const regexBytes =
+        "/" + newBytes.slice(0, 18) + "." + newBytes.slice(19) + "/";
+      nockCall = nockCall.replace(
+        new RegExp('"' + body + '"', "g"),
+        regexBytes
+      );
+    }
```
Comment (evantahler) [inline plugins/@grouparoo/snowflake/__tests__/utils/nockHelper.ts:97]: Dang!   That's a crazy API snowflake has.  I guess... they don't trust HTTP compression?

File: plugins/@grouparoo/snowflake/src/lib/connect.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    account,
+    username,
+    password,
+    database,
+    schema,
+    warehouse,
+  });
+
+  // @ts-ignore Set this on here for others to use
+  client.schemaName = schema;
```
Comment (evantahler) [inline plugins/@grouparoo/snowflake/src/lib/connect.ts]: just a note that you can do this to bypass the TS warnings:

```ts
client['schemaName'] = schema;
```

**Review by evantahler — CHANGES_REQUESTED** [review]
There's one thing to fix in the redshift plugin.

I really like the seperation/organization you've got here.  Perhaps in the near future there's a refactor in which we make a `@grouparoo/database-base` plugin which has a lot of these methods which PG/mySQL,Snowflake,etc can extend?

**Comment by bleonard** [comment]
> I really like the seperation/organization you've got here. Perhaps in the near future there's a refactor in which we make a @grouparoo/database-base plugin which has a lot of these methods which PG/mySQL,Snowflake,etc can extend?

Yes, I wanted to discuss with you the "core" or "base-plugin" thoughts for this source code, some stuff in destinations, and generic stuff (caching).

**Review by evantahler — APPROVED** [review]

**Comment by evantahler** [comment]
Closes https://github.com/grouparoo/grouparoo/issues/652

**Pull request merged**